chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc

```diff
@@ -0,0 +1,219 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/views/tabs/tab_drag_controller.h"
+
+#define TabDragController TabDragControllerChromium
+#include "src/chrome/browser/ui/views/tabs/tab_drag_controller.cc"
+#undef TabDragController
+
+#include "brave/browser/ui/views/tabs/features.h"
+
+TabDragController::~TabDragController() = default;
+
+// In order to use inner class defined in tab_drag_controller.cc, this function
+// is defined here.
+void TabDragController::Init(TabDragContext* source_context,
+                             TabSlotView* source_view,
+                             const std::vector<TabSlotView*>& dragging_views,
+                             const gfx::Point& mouse_offset,
+                             int source_view_offset,
+                             ui::ListSelectionModel initial_selection_model,
+                             EventSource event_source) {
+  if (!tabs::features::ShouldShowVerticalTabs()) {
+    TabDragControllerChromium::Init(source_context, source_view, dragging_views,
+                                    mouse_offset, source_view_offset,
+                                    initial_selection_model, event_source);
+    return;
+  }
+
+  // Most of code here is borrowed from TabDragControllerChromium::Init() but we
```

Comment:
same as above, we cannot copy code from upstream because it will get out of date and cause problems.